diagnostics/app/controllers/DiagnosticsController.scala

```diff
@@ -40,9 +40,17 @@ object DiagnosticsController extends Controller with Logging {
     TinyResponse.noContent()
   }
 
-  def cssOptions = postOptions
+  def csp = Action(jsonParser) { implicit request =>
+    val shouldReport = (new Instant().getMillis) % 10000 == 1
```

Comment:
haha, that was my fault, I suggested to use Int instead of double and then I wasn't sure how fast scala.util.Random would be instead of time.  It doesn't matter how evenly distributed it is, because we're not relying on that for the sample distribution, just for the sample size.  Unless some pages are more likely to be hit in a certain millisecond than others?
If the performance of scala.util.Random is good (I think it's pseudo random rather than using entropy) I wouldn't argue too strongly either way